### PR TITLE
dub: 1.23.0 -> 1.30.0

### PR DESCRIPTION
**Pull request by beardhatcode**
Bump dub to latest version

https://github.com/dlang/dub/releases/tag/v1.30.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

#### Note

dub 1.31.1 is also out, but using it breaks Literate (pkgs/development/tools/literate-programming/Literate/default.nix)

<details><summary>Build log of literate with dub 1.31.1</summary>

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/8r9ch4bl2p3dkgc50ykid9mxszizw3d5-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: SHELL=/nix/store/561wgc73s0x1250hrgp7jm22hhv7yfln-bash-5.2-p15/bin/bash
dub --root=lit/tangle build
    Starting Performing "debug" build using ldc2 for x86_64.
Error  /homeless-shelter: Permission denied
make: *** [Makefile:11: bin/tangle] Error 2

```

</details>

### CODE REVIEW

**Comment by beardhatcode** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>Literate</li>
    <li>btdu</li>
    <li>dub</li>
  </ul>
</details>

**Comment by beardhatcode** [comment]
dub 1.31.1 is also out, but using it breaks Literate (pkgs/development/tools/literate-programming/Literate/default.nix)

File: pkgs/development/tools/build-managers/dub/default.nix [hunk 1]
```diff
@@ -4,26 +4,26 @@ assert dcompiler != null;
...
 
   src = fetchFromGitHub {
     owner = "dlang";
     repo = "dub";
     rev = "v${version}";
-    sha256 = "06a4whsl1m600k096nwif83n7za3vr7pj1xwapncy5fcad1gmady";
+    sha256 = "sha256-iVl7bjblvIxvrUX7Phq6h4AIAmZjNVkGYYFA1hhsE7c=";
   };
 
   postUnpack = ''
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/dub/default.nix:18]:
```suggestion
  postPatch = ''
```

File: pkgs/development/tools/build-managers/dub/default.nix [hunk 2]
```diff
@@ -4,26 +4,26 @@ assert dcompiler != null;
...
-      patchShebangs .
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/dub/default.nix:19]: Can we narrow down what needs patching?

Comment (beardhatcode) [reply pkgs/development/tools/build-managers/dub/default.nix]: Done

**Pull request merged**